### PR TITLE
[Remote Excel support] Wrapper API for readxl::read_excel and excel_sheets

### PR DESCRIPTION
**Pull request by hidekoji**
### Description

Created wrapper API for

a) readxl::read_excel
b) readxl::excel_sheets

Update following API for remote RDATA files support

c) exploratory::getObjectListFromRdata
d) exploratory::getObjectFromRdata

### Checklist

Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test() 
- [ ] Tested with Exploratory

### CODE REVIEW

File: R/system.R [hunk 1]
```diff
@@ -1089,17 +1089,41 @@ select_columns <- function(x, ...) {
   return (df)
 }
 
+clear_cach_file <- function(url){
```
Comment (hideaki) [inline R/system.R]: Typo "cach".
Also, some comment about how this is used (to cache downloaded remote file etc.),
how it works (uses tempfile https://stat.ethz.ch/R-manual/R-devel/library/base/html/tempfile.html , and a R variable with name of hashed url is assigned to the path given by tempfile.)

File: R/system.R [hunk 2]
```diff
@@ -1089,17 +1089,41 @@ select_columns <- function(x, ...) {
...
+  hash <- digest::digest(url, "md5", serialize = FALSE)
+  tryCatch({
+    do.call(rm, c(as.name(hash)),envir = .GlobalEnv)
```
Comment (hideaki) [inline R/system.R]: I think it is better to call unlink() to remove the file. The tempfile page says that is the way to remove file.

File: R/system.R [hunk 3]
```diff
@@ -1089,17 +1089,41 @@ select_columns <- function(x, ...) {
...
 download_excel_file <- function(url){
-  ext <- stringr::str_to_lower(tools::file_ext(url))
-  # if no extension, assume the file extension as xlsx
-  if(ext == ""){
-    ext = "xlsx"
+  shouldCacheExcel <- getOption("tam.should.cache.excel")
+  filepath <- NULL
+  hash <- digest::digest(url, "md5", serialize = FALSE)
+  tryCatch({
+   filepath <- eval(as.name(hash))
```
Comment (hideaki) [inline R/system.R]: Indent is off here.

File: R/system.R [hunk 4]
```diff
@@ -1089,17 +1089,41 @@ select_columns <- function(x, ...) {
...
+  }, error = function(e){
+    filepath <- NULL
```
Comment (hideaki) [inline R/system.R]: Some comment would help on this error case about what kind of possible error this is handling.

File: R/system.R [hunk 5]
```diff
@@ -1089,17 +1089,41 @@ select_columns <- function(x, ...) {
...
+  })
+  if(!is.null(shouldCacheExcel) && isTRUE(shouldCacheExcel) && !is.null(filepath)){
+    filepath
```
Comment (hideaki) [inline R/system.R]: I guess this is cache hit case, but some comment would help to be able to figure it out quickly.

File: R/system.R (hunk 3 above)
Comment (hidekoji) [inline R/system.R]: fixed indent.

File: R/system.R (hunk 5 above)
Comment (hidekoji) [inline R/system.R]: added comment.

**Pull request merged**